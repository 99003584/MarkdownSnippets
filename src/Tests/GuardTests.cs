using System;
using ApprovalTests;
using Xunit;

public class GuardTests : TestBase
{
    [Fact]
    public void DirectoryIsFullyQualified_Invalid()
    {
        var argumentException = Assert.Throws<ArgumentException>(() => Guard.DirectoryIsFullyQualified("foo", "arg"));
        Approvals.Verify(argumentException.Message);
    }

    [Fact]
    public void DirectoryIsFullyQualified_Valid()
    {
        Guard.DirectoryIsFullyQualified("c:/", "arg");
    }
}