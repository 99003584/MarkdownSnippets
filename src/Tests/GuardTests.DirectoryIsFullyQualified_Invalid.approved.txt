Directory path must be fully qualified. Path: foo
Parameter name: arg